scripts/qa_crowbarsetup.sh

```diff
@@ -2958,6 +2959,74 @@ function glance_image_get_id()
     echo $image_id
 }
 
+function oncontroller_tempest_cleanup()
+{
+    if iscloudver 5plus; then
+        if tempest help cleanup; then
+            tempest cleanup --delete-tempest-conf-objects
+        else
+            /usr/bin/tempest-cleanup --delete-tempest-conf-objects || :
+        fi
+    else
+        /var/lib/openstack-tempest-test/bin/tempest_cleanup.sh || :
+    fi
+}
+
+function oncontroller_tempest_legacy()
+{
+    local image_name="SLES11-SP3-x86_64-cfntools"
+
+    # Upload a Heat-enabled image
+    if ! glance_image_exists $image_name; then
+        curl -s \
+            http://$clouddata/images/${image_name}.qcow2 | \
+            openstack image create \
+                --public --disk-format qcow2 --container-format bare \
+                --property hypervisor_type=kvm \
+                $image_name | tee glance.out
+    fi
+    local imageid=$(glance_image_get_id $image_name)
+    crudini --set /etc/tempest/tempest.conf orchestration image_ref $imageid
+    # test if is cnftools image prepared for tempest
+    wait_for 300 5 \
+        'openstack image show $imageid | grep active &>/dev/null' \
+        "prepare cnftools image"
+    pushd /var/lib/openstack-tempest-test
+    echo 1 > /proc/sys/kernel/sysrq
+    if iscloudver 5plus; then
+        if tempest help cleanup; then
+            tempest cleanup --init-saved-state
+        else
+            /usr/bin/tempest-cleanup --init-saved-state || :
+        fi
+    fi
+    ./run_tempest.sh -N $tempestoptions 2>&1 | tee tempest.log
+    local tempestret=${PIPESTATUS[0]}
+    testr last --subunit | subunit-1to2 > tempest.subunit.log
+
+    oncontroller_tempest_cleanup
+    popd
+    return $tempestret
+}
+
+function oncontroller_tempest()
+{
+    local tmode=${1:-smoke}
+    safely cd /var/lib/openstack-tempest-test
+    safely testr init
+    local datestr=$(date +%Y-%m-%d_%H%M%S)
+    local tlogfile=tempest_${mode}_${datestr}.log
+    local tempetoptions=$tmode
+    case $tmode in
+        smoke) tempetoptions="-N -t -s" ;;
+        full)  tempetoptions="-N -t"    ;;
+    esac
+    ./run_tempest.sh $tempestoptions 2>&1 | tee $tlogfile
+    local tempestret=${PIPESTATUS[0]}
```

Comment:
as pointed in my other comment, and in dirk's one, this is missing:
`testr last --subunit | subunit-1to2 > tempest.subunit.log`